esp-hal/src/ecc.rs

```diff
@@ -76,18 +81,35 @@ pub enum WorkMode {
     ModDiv                  = 11,
 }
 
-impl<'d> Ecc<'d> {
-    pub fn new(ecc: impl Peripheral<P = ECC> + 'd) -> Self {
+impl<'d> Ecc<'d, crate::Blocking> {
+    /// Create a new instance in [crate::Blocking] mode.
+    ///
+    /// Optionally an interrupt handler can be bound.    
+    pub fn new(ecc: impl Peripheral<P = ECC> + 'd, interrupt: Option<InterruptHandler>) -> Self {
         crate::into_ref!(ecc);
 
         PeripheralClockControl::enable(PeripheralEnable::Ecc);
 
+        if let Some(interrupt) = interrupt {
+            unsafe {
+                crate::interrupt::bind_interrupt(
+                    crate::peripherals::Interrupt::ECC,
+                    interrupt.handler(),
+                );
+                crate::interrupt::enable(crate::peripherals::Interrupt::ECC, interrupt.priority())
+                    .unwrap();
+            }
+        }
+
         Self {
             ecc,
             alignment_helper: AlignmentHelper::default(),
+            phantom: PhantomData,
         }
     }
+}
 
+impl<'d, DM: crate::Mode> Ecc<'d, DM> {
     pub fn free(self) -> PeripheralRef<'d, ECC> {
```

Comment:
Wasn't aware we still had `free` methods 😅, could you remove this?

Reply:
Oh, didn't notice that one ... sure